drivers/ethernet/eth_lan9250.c

```diff
@@ -0,0 +1,709 @@
+				      (ctx->mac_address[2] << 16) | (ctx->mac_address[3] << 24));
+	lan9250_write_mac_reg(dev, LAN9250_HMAC_ADDRH,
+			      ctx->mac_address[4] | (ctx->mac_address[5] << 8));
+
+	return 0;
+}
+
+static int lan9250_spi_check(const struct device *dev)
+{
+	uint32_t tmp;
+	uint8_t retries = LAN9250_DEFAULT_NUMOF_RETRIES;
+
+	do {
+		lan9250_read_sys_reg(dev, LAN9250_BYTE_TEST, &tmp);
+		k_busy_wait(USEC_PER_MSEC * 1U);
+	} while (tmp != LAN9250_BYTE_TEST_DEFAULT && retries--);
+
+	return 0;
+}
+
+static int lan9250_hw_cfg_check(const struct device *dev)
+{
+	uint32_t tmp;
+
+	do {
+		lan9250_read_sys_reg(dev, LAN9250_HW_CFG, &tmp);
+		k_busy_wait(USEC_PER_MSEC * 1U);
+	} while ((tmp & LAN9250_HW_CFG_DEVICE_READY) == 0);
+
+	return 0;
+}
+
+static int lan9250_sw_reset(const struct device *dev)
+{
+	/* Wait until LAN9250 SPI bus is ready */
+	lan9250_wait_ready(dev, LAN9250_BYTE_TEST, BOTR_MASK, LAN9250_BYTE_TEST_DEFAULT,
+			   LAN9250_RESET_TIMEOUT);
+	lan9250_write_sys_reg(dev, LAN9250_RESET_CTL,
+			      LAN9250_RESET_CTL_HMAC_RST | LAN9250_RESET_CTL_PHY_RST |
+				      LAN9250_RESET_CTL_DIGITAL_RST);
+
+	/* Wait until LAN9250 SPI bus is ready */
+	lan9250_wait_ready(dev, LAN9250_BYTE_TEST, BOTR_MASK, LAN9250_BYTE_TEST_DEFAULT,
+			   LAN9250_RESET_TIMEOUT);
+
+	return 0;
+}
+
+static int lan9250_configure(const struct device *dev)
+{
+	uint32_t tmp;
+
+	lan9250_hw_cfg_check(dev);
+
+	/* Read LAN9250 hardware ID */
+	lan9250_read_sys_reg(dev, LAN9250_ID_REV, &tmp);
+	if ((tmp & LAN9250_ID_REV_CHIP_ID) != LAN9250_ID_REV_CHIP_ID_DEFAULT) {
+		LOG_ERR("ERROR: Bad Rev ID: %08x\n", tmp);
+		return -ENODEV;
+	}
+
+	/* Configure TX FIFO size mode to be 8:
+	 *
+	 *   - TX data FIFO size:   7680
+	 *   - RX data FIFO size:   7680
+	 *   - TX status FIFO size: 512
+	 *   - RX status FIFO size: 512
+	 */
+	lan9250_write_sys_reg(dev, LAN9250_HW_CFG,
+			      LAN9250_HW_CFG_MBO | LAN9250_HW_CFG_TX_FIF_SZ_8KB);
+
+	/* Configure MAC automatic flow control:
+	 *
+	 *  Reference: Microchip Ethernet LAN9250
+	 *  https://github.com/microchip-pic-avr-solutions/ethernet-lan9250/
+	 *  LAN_Regwrite32(AFC_CFG, 0x006E3741);
+	 *
+	 */
+	lan9250_write_sys_reg(dev, LAN9250_AFC_CFG, 0x006e3741);
+
+	/* Configure interrupt:
+	 *
+	 *   - Interrupt De-assertion interval: 100
+	 *   - Interrupt output to pin
+	 *   - Interrupt pin active output low
+	 *   - Interrupt pin push-pull driver
+	 */
+	lan9250_write_sys_reg(dev, LAN9250_IRQ_CFG,
+			      LAN9250_IRQ_CFG_INT_DEAS_100US | LAN9250_IRQ_CFG_IRQ_EN |
+				      LAN9250_IRQ_CFG_IRQ_TYPE_PP);
+
+	/* Configure interrupt trigger source, please refer to macro
+	 * LAN9250_INT_SOURCE.
+	 */
+	lan9250_write_sys_reg(dev, LAN9250_INT_EN,
+			      LAN9250_INT_EN_PHY_INT_EN | LAN9250_INT_EN_RSFL_EN);
+
+	/* Disable TX data FIFO available interrupt */
+	lan9250_write_sys_reg(dev, LAN9250_FIFO_INT,
+			      LAN9250_FIFO_INT_TX_DATA_AVAILABLE_LEVEL |
+				      LAN9250_FIFO_INT_TX_STATUS_LEVEL);
```

Comment:
Indentation is wrong here and in many similar statements below

Reply:
done :)